airflow/providers/airbyte/operators/parse.py

```diff
@@ -0,0 +1,271 @@
+                    name=schema,
+                    field=field_name,
+                )
+            ],
+            transformationType="IDENTITY",
+            transformationDescription="",
+        )
+
+    return ColumnLineageDatasetFacet(fields=fields)
+
+
+def resolve_metadata(
+    connection: dict,
+    destination: Dict[str, Any],
+    namespace: str,
+) -> AirbyteSyncDefinitions:
+    streams = get_streams(connection)
+
+    definitions = []
+
+    prefix = connection.get("prefix", "")
+
+    for stream_data in streams:
+        stream = stream_data.get("stream", None)
+        # skip if stream is missing
+        if stream is None:
+            continue
+
+        stream_name = stream.get("name", None)
+        if stream_name is None:
+            continue
+
+        input_schema = stream.get("namespace", "")
+
+        resolved_schema = resolve_table_schema(
+            name_source=connection.get("namespaceDefinition", "source"),
+            source=stream,
+            destination=destination,
+            connection=connection,
+        )
+
+        # skip if jsonSchema is missing
+        json_schema = stream.get("jsonSchema", None)
+        if json_schema is None:
+            continue
+
+        properties = json_schema.get("properties", None)
+        if properties is None:
+            continue
+
+        schema = map_airbyte_type_to_ol_schema_field(properties)
+        target_table_name = prefix + stream_name
+
+        column_lineage = resolve_column_lineage(
+            namespace=namespace,
+            schema=resolved_schema,
+            stream=properties,
+        )
+
+        definitions.append(
+            AirbyteSyncDefinition(
+                target_facet=TargetFacets(
+                    schema=SchemaDatasetFacet(fields=schema), column_lineage=column_lineage
+                ),
+                source_facet=SourceFacets(
+                    schema=SchemaDatasetFacet(fields=schema),
+                ),
+                source_table=f"{input_schema}.{stream_name}" if input_schema != "" else stream_name,
+                target_table=(
+                    f"{resolved_schema}.{target_table_name}" if resolved_schema != "" else target_table_name
+                ),
+            )
+        )
+
+    return AirbyteSyncDefinitions(definitions=definitions, namespace=namespace)
+
+
+def get_field_type(field_metadata: Dict[str, Any]) -> str:
+    airbyte_field_type = field_metadata.get("type", None)
+
+    airbyte_target_field_type = ""
+
+    if isinstance(airbyte_field_type, list):
+        airbyte_target_field_type = map_airbyte_types(airbyte_field_type, field_metadata.get("format", None))
+
+    if isinstance(airbyte_field_type, str):
+        airbyte_target_field_type = map_airbyte_type(airbyte_field_type, field_metadata.get("format", None))
+
+    return airbyte_target_field_type
+
+
+def map_airbyte_type_to_ol_schema_field(stream: Dict[str, Any]) -> List[SchemaDatasetFacetFields]:
```

Comment:
It's quite complex and hard to read, can we split it into separate functions somehow?